src/core/types/execution_context.ts

```diff
@@ -16,11 +16,13 @@ export type KibanaExecutionContext = {
   /**
    * Kibana application initated an operation.
    * */
-  readonly type: string; // 'visualization' | 'actions' | 'server' | ..;
-  /** public name of a user-facing feature */
-  readonly name: string; // 'TSVB' | 'Lens' |  'action_execution' | ..;
+  readonly type?: string; // 'visualization' | 'actions' | 'server' | ..;
+  /** public name of an application or a user-facing feature */
+  readonly name?: string; // 'TSVB' | 'Lens' |  'action_execution' | ..;
+  /** a stand alone, logical unit such as an application page or tab */
+  readonly page?: string;
   /** unique value to identify the source */
-  readonly id: string;
+  readonly id?: string;
```

Comment:
What is the problem of keeping `id` as a required field? Can we set it to any meaningful value?

Reply:
Actually `id` is not required - for example on a dashboards list there's no entity to identify.

Reply:
> Actually id is not required - for example on a dashboards list there's no entity to identify.

Do you mean `any` application page without a unique identifier? Isn't page name a unique identifier in this case?